src/storage/data_table.cpp

```diff
@@ -7,6 +9,8 @@
 
 namespace terrier::storage {
 DataTable::DataTable(BlockStore *store, const BlockLayout &layout) : block_store_(store), accessor_(layout) {
+  data_table_counter_ = new DataTableCounter;
```

Comment:
Why not inline these counters? Do we support inlining these counters in the class themselves?